### ISSUE TITLE
Not working on VirtualBox and VMware

### ISSUE THREAD

**Issue by Ascript89** [jdah/tetris-os#19]
I also saw another issue commenting it doesn't works on VirtualBox, so I wanted to test it on VMware (Workstation 16), and it was the same result. Why can I only run this on Qemu?

**Comment by farmerbb** [comment]
Have you tried compiling it with `ENABLE_MUSIC` removed from main.c?

**Comment by KingUndeadCodes** [comment]
Maybe because testing was only done in QEMU?
Like @farmerbb compile it yourself. 👍

**Comment by jdah** [comment]
Please specify what you meana by "not working". Does it boot? Does it get stuck somewhere?

**Comment by fixator10** [comment]
VMWare 15.5.6, "Other" VM without hard drive:
![image](https://user-images.githubusercontent.com/11073934/115955013-32c31f80-a505-11eb-8669-ad1dcd511bea.png)

Tried CD/DVD drive in SATA, IDE and SCSI nodes

Using precompiled iso from releases

**Comment by fixator10** [comment]
After changing file extension to img and adding image as floppy disk (thanks #22 for idea) - that came out:
![image](https://user-images.githubusercontent.com/11073934/115955562-4459f680-a508-11eb-8997-62ed8df2e8c9.png)

**Comment by Ascript89** [comment]
> Please specify what you meana by "not working". Does it boot? Does it get stuck somewhere?

@jdah it just doesn't recognize it as an OS, like if the bootable virtual cd isn't valid.
After looking at some comments, I think the best will be compiling it myself.

**Comment by zment4** [comment]
The resulting iso file isn't actually a real ISO file, just a flat, raw binary file. Also, DISK ERROR suggests that LBA addressing reads don't work. You could try the CHS read PR #44 or a release boot.bin with that compiled in from my [fork](https://github.com/zment4/tetris-os/releases/pcspkr_v1)

**Comment by zment4** [comment]
Got it working on VirtualBox with the CHS reads #44 

![image](https://user-images.githubusercontent.com/2785557/116317033-7b0b6780-a7bb-11eb-8e30-4140cd1ec47b.png)

Can't get sound working though. Also needed to rename the file to .img and use it as a floppy disk to get it to load.

**Comment by jdah** [comment]
Closing this since CHS reads have been merged in, thanks to @zment4 !

**Comment by zment4** [comment]
> Can anyone test with the new dynamic SB16 code, etc.?
I tested both of those with the official build as well as my fork. Couldn't get either PC Speaker or SB16 working (VirtualBox doesn't even have PC Speaker emulation, and I don't know how you enable PC Speaker in VMWare)

**Comment by fixator10** [comment]
Same VMWare, no sound with previously mentioned SB16 config options and VM basically is unkillable. I cant even power off this VM.

**Comment by noxwaste** [comment]
I'm using VirtualBox 6.1.16 on Windows 10 Pro x64 and the .img booted from floppy works perfectly fine, along with sound. No issues at all. Just for reference, these are my VM settings:

Base memory: 256 MB
Chipset: PIIX3
Processor: Enable PAE/NX (only 1 processor assigned to VM)
Display: 9 MB video memory
Storage: Floppy controller with boot.img mounted in it
IDE controller with Tetris-OS.vdi on that (512 MB image size - could probably be 500x smaller)
Audio: Windows DirectSound as Host Audio Driver, SoundBlaster 16 as Audio Controller

Everything else is pretty much default.

**Comment by pakair** [comment]
followed the VB settings, it plays with sound although there is a hissing sound here and there, not sure where it comes from.

**Comment by limmers2015** [comment]
I got it running in about 10min on VirtualBox, you just have to tell it that it's a floppy.

**Comment by pakair** [comment]
@limmers2015 did you use the same settings as @noxwaste posted a few days earlier?

**Comment by limmers2015** [comment]
Nope, I only found this issue when I was looking to start one asking where to go to modify the render size to be larger.

Here's my settings:
![VirtualBox_TetrOS_Settings](https://user-images.githubusercontent.com/34211966/116952929-f193e380-ac59-11eb-8e06-0e82b34e694c.png)
